### PR TITLE
Improve Template#inspect output

### PR DESCRIPTION
**Pull request by jhawthorn**
This is a quality of life improvement for those developing ActionView itself (cc @tenderlove)

This PR first renames the existing `inspect` implementation to `short_identifier`. This improves the 👀 reading of `identifier_method_name`, which used to read `inspect.tr("^a-z_", "_")`, which is a bit awkward because it relies on what `inspect` _happens_ to return. `short_identifier.tr("^a-z_", "_")` conveys what it does better (and allows us to change `inspect` 🎉 ).

Inspect used to return a string like `app/views/home/index.html.erb`, which is confusing because I would normally expect to see a type and it isn't immediately obvious that that isn't a string, particularly when in an array (ie. `[some/template, another/template]` parses to me the same as `["some/template", "another/template"]` on first glance).

This PR changes inspect to return a string like:
```
#<ActionView::FileTemplate app/views/tags/_tag.html.erb locals=["name"]>
```

Including the class name makes it look more like most `#inspect` methods. I also added `locals` to this output because it's often different for the same identifier (we need to compile each set of locals separately). It _could_ be helpful to include handler, formats, and variants, but they should be inferable by the developer from the identifier, so I didn't include them.

I'd guess the reason we override inspect here in the first place is to avoid printing out the whole source code for the template.

### CODE REVIEW

File: actionview/lib/action_view/template.rb [hunk 1]
```diff
@@ -202,8 +202,12 @@ def refresh(view)
       end
     end
 
+    def short_identifier
+      @short_identifier ||= defined?(Rails.root) ? identifier.sub("#{Rails.root}/", "") : identifier
```
Comment (kaspth) [inline actionview/lib/action_view/template.rb]: I’m guessing what’s identifier and short_identifier here is what’s path(?) and virtual_path elsewhere in Action View. Might be nice if the naming was unified for our internals. Though that’s another PR.

**Review by kaspth — APPROVED** [review]
Do we print the inspect output in logs or the log subscriber? Could be good to double check what a rendering log output looks like.

**Comment by jhawthorn** [comment]
> Do we print the inspect output in logs or the log subscriber? Could be good to double check what a rendering log output looks like.

👍 Good idea. Logging looks unchanged in my test app. I also tested an exception from a view and it looked normal.

screen shot 2019-02-25 at 14 33 30

File: actionview/lib/action_view/template.rb (hunk 1 above)
Comment (jhawthorn) [reply actionview/lib/action_view/template.rb]: I think they are a little different. `virtual_path` is relative to the view directory (ex. `app/views`), where `short_identifier` is relative to the app root (or from `/` if from a gem outside). I think it's also possible for identifier not to be from the FS if the template comes from some other source.

I do agree that it does feel like something here could be unified though!

**Comment by kaspth** [comment]
Gotcha! Thanks for checking both.

**Pull request merged**